e2e/nfs.go

```diff
@@ -0,0 +1,564 @@
+	resources := []ResourceDeployer{
+		&yamlResource{
+			filename:     nfsDirPath + csiDriverObject,
+			allowMissing: true,
+		},
+		&yamlResource{
+			filename:     examplePath + cephConfconfigMap,
+			allowMissing: true,
+		},
+		&yamlResourceNamespaced{
+			filename:  nfsDirPath + nfsProvisionerRBAC,
+			namespace: cephCSINamespace,
+		},
+		&yamlResourceNamespaced{
+			filename:  nfsDirPath + nfsProvisionerPSP,
+			namespace: cephCSINamespace,
+		},
+		&yamlResourceNamespaced{
+			filename:   nfsDirPath + nfsProvisioner,
+			namespace:  cephCSINamespace,
+			oneReplica: true,
+		},
+		&yamlResourceNamespaced{
+			filename:  nfsDirPath + nfsNodePluginRBAC,
+			namespace: cephCSINamespace,
+		},
+		&yamlResourceNamespaced{
+			filename:  nfsDirPath + nfsNodePluginPSP,
+			namespace: cephCSINamespace,
+		},
+		&yamlResourceNamespaced{
+			filename:  nfsDirPath + nfsNodePlugin,
+			namespace: cephCSINamespace,
+		},
+		&rookNFSResource{
+			f:           f,
+			modules:     []string{"rook", "nfs"},
+			orchBackend: "rook",
+		},
+		&yamlResourceNamespaced{
+			filename:  nfsExamplePath + nfsRookCephNFS,
+			namespace: rookNamespace,
+		},
+	}
+
+	for _, r := range resources {
+		err := r.Do(action)
+		if err != nil {
+			e2elog.Failf("failed to %s resource: %v", action, err)
+		}
+	}
+}
+
+func createNFSStorageClass(
+	c clientset.Interface,
+	f *framework.Framework,
+	enablePool bool,
+	params map[string]string) error {
+	scPath := fmt.Sprintf("%s/%s", nfsExamplePath, "storageclass.yaml")
+	sc, err := getStorageClass(scPath)
+	if err != nil {
+		return err
+	}
+	sc.Parameters["nfsCluster"] = "my-nfs"
+	sc.Parameters["server"] = "rook-ceph-nfs-my-nfs-a." + rookNamespace + ".svc.cluster.local"
+
+	// standard CephFS parameters
+	sc.Parameters["fsName"] = fileSystemName
+	sc.Parameters["csi.storage.k8s.io/provisioner-secret-namespace"] = cephCSINamespace
+	sc.Parameters["csi.storage.k8s.io/provisioner-secret-name"] = cephFSProvisionerSecretName
+
+	sc.Parameters["csi.storage.k8s.io/controller-expand-secret-namespace"] = cephCSINamespace
+	sc.Parameters["csi.storage.k8s.io/controller-expand-secret-name"] = cephFSProvisionerSecretName
+
+	sc.Parameters["csi.storage.k8s.io/node-stage-secret-namespace"] = cephCSINamespace
+	sc.Parameters["csi.storage.k8s.io/node-stage-secret-name"] = cephFSNodePluginSecretName
+
+	if enablePool {
+		sc.Parameters["pool"] = "myfs-replicated"
+	}
+
+	// overload any parameters that were passed
+	if params == nil {
+		// create an empty params, so that params["clusterID"] below
+		// does not panic
+		params = map[string]string{}
+	}
+	for param, value := range params {
+		sc.Parameters[param] = value
+	}
+
+	// fetch and set fsID from the cluster if not set in params
+	if _, found := params["clusterID"]; !found {
+		var fsID string
+		fsID, err = getClusterID(f)
+		if err != nil {
+			return fmt.Errorf("failed to get clusterID: %w", err)
+		}
+		sc.Parameters["clusterID"] = fsID
+	}
+	sc.Namespace = cephCSINamespace
```

Comment:
sc doesnot need namespace

Reply:
right, it is just a copy/paste of the RBD or CephFS StorageClass creation. I'll clean those up in a separate commit (in this PR).